vsts/scripts/pullandretag.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT license.
+# --------------------------------------------------------------------------------------------
+
+set -o pipefail
+
+while read sourceImage; do
+  # Always use specific build number based tag and then use the same tag to create a 'latest' tag and push it
+  if [[ $sourceImage != *:latest ]]; then
+    echo "Pulling the source image $sourceImage ..."
+    docker pull "$sourceImage" | sed 's/^/     /'
+        
+    # Trim the build number and append the '':latest' to end of it
+    newtag="${sourceImage%:*}:latest"
+    echo
+    echo "Tagging the source image with tag $newtag ..."
+    docker tag "$sourceImage" "$newtag" | sed 's/^/     /'
```

Comment:
I _think_ we might already been doing this but could you check if we untag the pulled image too...just to be clear

1. pull `foo:10.1`
2. retag `foo:10.1` to `foo:latest`
3. untag `foo:10.1` (this is important since there would bunch of them lying on the agent)

But I _think_ we might already do the untagging in `vsts/scripts/dockerCleanup.sh` with the following line, but please check again:
`UntagImages "oryxdevmcr.azurecr.io/public/oryx/*:*.*"`

Reply:
I think this is already being done. So you can ignore it!